tests/unit/test_import.py

```diff
@@ -0,0 +1,26 @@
+import importlib
+import sys
+from unittest.mock import patch
+
+import pytest
+
+import pybop
+
+
+@pytest.mark.unit
+def test_multiprocessing_init_non_win32(monkeypatch):
+    """Test multiprocessing init on non-Windows platforms"""
+    monkeypatch.setattr(sys, "platform", "linux")
+    print(sys.platform)
+    with patch("multiprocessing.set_start_method") as mock_set_start_method:
+        importlib.reload(pybop)
```

Comment:
[`importlib.reload()`](https://docs.python.org/3/library/importlib.html#importlib.reload) does not reload all sub-modules of a package and usually caches results a lot. I would instead recommend removing `pybop` and any `pybop.*` imports from the `sys.modules` dictionary, and then using `importlib.import_module("pybop")` to import it again and achieve the desired behaviour.